.circleci/check-testnet-deployment.js

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env node
+
+const { waitUntilSync, getBlockHeight, getCommit } = require('@orbs-network/orbs-nebula/lib/metrics');
+
+const topology = require('./config.json');
+const TARGET_HASH = process.argv[2];
+
+if (!TARGET_HASH) {
+    console.log('No target hash to check');
+    process.exit(1);
+}
+
+async function eventuallyDeployed({ chainId, nodes }) {
+    // First let's poll the nodes for the correct version
+    // Until we meet the correct version on all nodes or we bail by the timeout defined
+    let versionDeployed = false;
+    let runsCount = 0;
+    const maxRetries = 30; // 15 minutes
+
+    do {
+        if (runsCount + 1 > maxRetries) {
+            console.log('Max retries limit reached for chainId: ', chainId);
+            break;
+        }
+
+        runsCount++;
+        console.log('Polling all nodes for their version on chainId ', chainId);
+        const results = await Promise.all(nodes.map(({ ip }) => {
+            return getCommit(`${ip}/vchains/${chainId}`);
+        }))
+            .catch(err => {
+                console.log('failed getting the commit hash of one of the nodes for some reason');
+                console.log('error provided:', err);
+            });
+
+        console.log('Got: ', results);
+        if (results.filter(version => version === TARGET_HASH).length === nodes.length) {
+            // Version updated on all nodes
+            versionDeployed = true;
+            continue;
+        }
+
+        console.log('Sleeping for 30 seconds..');
+        await new Promise((r) => { setTimeout(r, 30 * 1000); });
+    } while (versionDeployed === false);
+
+    return {
+        ok: versionDeployed
+    };
+}
+
+async function eventuallyClosingBlocks({ chainId, nodes }) {
```

Comment:
Same can be done with nebula call for status, waiting for all of them to become green.